### ISSUE TITLE
document xgboost YARN build parameters

### ISSUE THREAD

**Issue by edi-bice** [dmlc/wormhole#50]
the wiki/readthedocs build instructions do not mention dmlc parameter - one finds out the hard way after running in cluster and looking at job logs.

furthermore dmlc=1 which is suggested in those job error logs doesn't cut it - the build fails with a cryptic "file does not exist 1/make/config.mk" ... which apparently is due to the "dmlc=1"

googling, and reading, around it appears one can/should specify a path to the dmlc-core via dmlc build parameter like so "make dmlc=../dmlc-core"

**Comment by tqchen** [comment]
Thanks @edi-bice to point this out, can you open a PR on this? Thanks!

**Comment by edi-bice** [comment]
Sure thing @tqchen 

https://github.com/dmlc/wormhole/pull/51

**Comment by tqchen** [comment]
Great! Thanks for the contribution. I definitely agree that we should make things in distributed xgboost easier, and your contribution is valuable

**Comment by edi-bice** [comment]
Another issue I struggled with was the dependency on environment variables particularly HADOOP_HDFS_HOME. Tried to set it manually in each box in the cluster! Luckily only have 10 boxes. Didn't work as it gets overridden somehow. Or different user. Tried to set it in Ambari (HDP Hortonworks Hadoop) and it broke the whole thing. Different Hadoop distributions apparently do wildly different things. In the end I replaced it with the actual path in the run_hdfs.py script. One and done.
